### PR TITLE
Add Modal Stack to Live Visual Tree

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###

Return Modal Pages as part of a Windows VisualTreeElements

Implements #1447

### CODE REVIEW

File: src/Controls/src/Core/Xaml/Diagnostics/VisualDiagnostics.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ internal static void OnChildAdded(Element parent, Element child)
 			if (child is null)
 				return;
 
-			var index = parent?.AllChildren.IndexOf(child) ?? -1;
+			var index = (parent as IVisualTreeElement)?.GetVisualChildren().IndexOf(child) ?? -1;
```
Comment (PureWeen) [inline src/Controls/src/Core/Xaml/Diagnostics/VisualDiagnostics.cs:30]: @mattleibow @drasticactions ?

Comment (drasticactions) [reply src/Controls/src/Core/Xaml/Diagnostics/VisualDiagnostics.cs:30]: Looks good to me. As long as the tests are good, I'm good. 👍

**Review by drasticactions — APPROVED** [review]

**Comment by PureWeen** [comment]
@drasticactions I'm not really fixing ListView here. The change just adjusts ListView to work the same as it did with previous structures.  `_logicalChildren` is a bit of a confusing name for the variable, because ListView doesn't return that for the base logicalchildren. It's purely a local variable. The `LogicalChildrenInternal` on ListView will still always be empty

Previously we had added
```C#
internal override IEnumerable<Element> ChildrenNotDrawnByThisElement => _logicalChildren;
```

In order to make it work with VisualDiagnostics but now that we switched to the new IVET structure we need to return `_logicalChildren` via that to maintain current behavior. 

This PR in no way makes any improvements on ListView :-)

**Review by mattleibow — APPROVED** [review]
Feels much better!

**Review by rmarinho — APPROVED** [review]

**Pull request merged**